File: api/src/test/java/net/blueshell/api/factory/model/GuestFactory.java
package net.blueshell.api.factory.model;

import com.github.javafaker.Faker;
import lombok.RequiredArgsConstructor;
import net.blueshell.api.model.event.Guest;
import org.springframework.stereotype.Component;

import java.util.UUID;
import java.util.concurrent.atomic.AtomicLong;

/**
 * Factory for Guest model test instances.
 */
@Component
@RequiredArgsConstructor
public class GuestFactory {

    private static final AtomicLong counter = new AtomicLong(1000);
    private final Faker faker;

    public Guest createBasic() {
        Guest guest = new Guest();
        guest.setId(generateId());
        guest.setName(faker.name().fullName());
        guest.setDiscord(faker.name().username() + "#" + faker.number().numberBetween(1000, 9999));
        guest.setEmail(faker.internet().emailAddress());
        guest.setPhoneNumber(faker.phoneNumber().phoneNumber());
        guest.setAccessToken(UUID.randomUUID().toString());
        return guest;
    }

    public Guest createFull() {
        return createBasic();
    }

    public Guest createWithCustomizations(java.util.function.Consumer<Guest> customizer) {
        Guest guest = createFull();
        customizer.accept(guest);
        return guest;
    }

    private Long generateId() {
        return counter.incrementAndGet();
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/model/event/GuestFactory.kt
package net.blueshell.api.factory.model.event

import com.github.javafaker.Faker
import net.blueshell.api.feature.event.model.Guest
import org.springframework.stereotype.Component
import java.util.UUID
import java.util.concurrent.atomic.AtomicLong
import java.util.function.Consumer

/**
 * Factory for Guest model test instances.
 */
@Component
class GuestFactory(
    private val faker: Faker
) {

    fun createBasic(): Guest {
        val guest = Guest()
        guest.name = faker.name().fullName()
        guest.discord = faker.name().username() + "#" + faker.number().numberBetween(1000, 9999)
        guest.email = faker.internet().emailAddress()
        guest.phoneNumber = faker.phoneNumber().phoneNumber()
        guest.accessToken = UUID.randomUUID().toString()
        return guest
    }

    fun createFull(): Guest = createBasic()

    fun createWithCustomizations(customizer: Consumer<Guest>): Guest {
        val guest = createFull()
        customizer.accept(guest)
        return guest
    }

    private fun generateId(): Long = COUNTER.incrementAndGet()

    private companion object {
        val COUNTER = AtomicLong(1000)
    }
}
